### ISSUE TITLE
Not compatible with toJSON instance method

### ISSUE THREAD

**Issue by Faleij** [valery-barysok/json-cycle#2]
Example:
```javascript
var jc = require("json-cycle")

class Baz {
    constructor() {
        this.foo = 'foo';
    }
    toJSON() {
        return 'bar';
    }
}

 // x should be: '"bar"' in both console.log below
console.log(JSON.stringify({ x: new Baz() }));       
console.log(JSON.stringify(jc.decycle({ x: new Baz() }))); // produces wrong output; x: '{"foo": "foo"}'
```
What happens in ``jc.decycle`` is that the deep copy of Baz does not have its prototype chain preserved, thus toJSON is never called in ``JSON.stringify``.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON()_behavior

**Comment by valery-barysok** [comment]
@Faleij, thanks for contribution. Can you make pull request to make it possible to release new version with your fix?

**Comment by Faleij** [comment]
@valery-barysok pull request is ready

**Comment by valery-barysok** [comment]
@Faleij, merged and published under 1.1.0 version
